### ISSUE TITLE
Support `reduction` param in other loss functions

### ISSUE THREAD

**Issue by tim-becker** [tinygrad/tinygrad#6247]
torch lets you control how the loss is reduced in most (all?) loss functions, which is convenient.

Currently tinygrad only supports this in the newly added `Tensor.cross_entropy`.

Note: I wanted this for `Tensor.sparse_categorical_crossentropy`, but was able to work around it by using `Tensor.cross_entropy` instead (with an appropriate reshaping).

Still worth tracking this issue in general though, IMO.

**Comment by geohot** [comment]
I'd merge a clean tested fix for this
